packages/core/test/financial-templates/perpetual-multiparty/PerpetualCreator.js

```diff
@@ -203,6 +205,66 @@ contract("PerpetualCreator", function(accounts) {
     assert.equal(await perpetual.timerAddress(), await perpetualCreator.timerAddress());
   });
 
+  it("Constructs new synthetic currency properly", async function() {
+    // Use non-18 decimal precision for collateral currency to test that synthetic matches precision.
+    collateralToken = await Token.new("UMA", "UMA", 8, { from: contractCreator });
+    constructorParams.collateralAddress = collateralToken.address;
+
+    // Whitelist collateral currency
+    await collateralTokenWhitelist.addToWhitelist(collateralToken.address, { from: contractCreator });
+
+    // Create new derivative contract.
+    let createdAddressResult = await perpetualCreator.createPerpetual(constructorParams, {
+      from: contractCreator
+    });
+    let perpetualAddress;
+    truffleAssert.eventEmitted(createdAddressResult, "CreatedPerpetual", ev => {
+      perpetualAddress = ev.perpetualAddress;
+      return ev.perpetualAddress != 0 && ev.deployerAddress == contractCreator;
+    });
+    let perpetual = await Perpetual.at(perpetualAddress);
+
+    // New synthetic currency and collateral currency should have the same precision.
+    const tokenCurrency = await Token.at(await perpetual.tokenCurrency());
+    const collateralCurrency = await Token.at(await perpetual.collateralCurrency());
+    assert.equal((await tokenCurrency.decimals()).toString(), (await collateralCurrency.decimals()).toString());
+
+    // New derivative contract holds correct permissions.
+    const tokenContract = await SyntheticToken.at(tokenCurrency.address);
+    assert.isTrue(await tokenContract.isMinter(perpetualAddress));
+    assert.isTrue(await tokenContract.isBurner(perpetualAddress));
+    assert.isTrue(await tokenContract.holdsRole(0, perpetualAddress));
+  });
+
+  it("If collateral currency does not implement the decimals() method then synthetic currency defaults to 18 decimals", async function() {
```

Comment:
As pointed else elsewhere what happens if the token does not have a `decimals()` method and is NOT 18 decimal points precicion?

Reply:
So what's interesting is that "precision" is just an abstraction. The token is still represented in `uint256`, but its a matter of the front ends/clients to deal with how the precision works. If a token does not have `decimals()` then it gets really difficult for clients to determine how to deal with the balance precision, so I guess the default is 18.